File: php/features.php
<?php
declare(strict_types=1);

// EuroRates SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EuroRatesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EuroRatesBaseFeature();
            case "test":
                return new EuroRatesTestFeature();
            default:
                return new EuroRatesBaseFeature();
        }
    }
}

File: py/features.py
# EuroRates SDK feature factory

from feature.base_feature import EuroRatesBaseFeature
from feature.test_feature import EuroRatesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EuroRatesBaseFeature(),
        "test": lambda: EuroRatesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
